### ISSUE TITLE
Disable publish on backend for collections behind on schema version

### ISSUE THREAD

**Issue by nayib-jose-gloria** [chanzuckerberg/single-cell-data-portal#4805]
Do not allow publish if private collection has dataset(s) with a schema_version less than the current latest

**Comment by nayib-jose-gloria** [comment]
Won't Do; taken care of by validation step blocking upload of previous schema versions + publish being blocked on collection versions with mismatching dataset scehma versions + migration moving anything in the corpus to the new schema version
